Updating parameters.

MahendraAgrawal opened the pull request:
Now these templates will work on both Azure/AzureStack.

azurecla commented:
Hi __@MahendraAgrawal__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (mahendra). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

MahendraAgrawal commented:
This seems strange. it works locally. let me try with hardcode unique values and see. This could be BOT error.

azurecla commented:
Hi __@MahendraAgrawal__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (mahendra). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

MahendraAgrawal commented:
Reopening it as discussed with Kay. The failures are expected in automated system because of custom image template. Please help merge these changes.

singhkays commented:
Adding @bmoore-msft @gatneil for review

gatneil commented on rds-deployment-custom-image-rdsh/azuredeploy.json:
please use raw strings for apiVersion instead of variables

gatneil commented on rds-deployment-custom-image-rdsh/azuredeploy.json:
Please see the best practice around _artifactsLocation and _artifactsLocationSasToken: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

gatneil commented on rds-deployment-custom-image-rdsh/azuredeploy.json:
are we sure we want a default password?

gatneil commented on rds-deployment-custom-image-rdsh/azuredeploy.json:
Shouldn't AdminCreds go in protectedSettings?

gatneil commented on rds-deployment-custom-image-rdsh/azuredeploy.parameters.json:
```diff
@@ -5,35 +5,17 @@
     },
     "dnsLabelPrefix": {
-      "value": "GEN-UNIQUE"
+      "value": "rduniquedns1"
```
please use GEN-UNIQUE or GEN-UNIQUE-[some-number]

gatneil commented on rds-deployment-custom-image-rdsh/azuredeploy.parameters.json:
```diff
@@ -5,35 +5,17 @@
-    },
     "rdshImageStorageAccountName": {
-      "value": "Provide-StorageName"
+      "value": "imageuniquestorename1"
```
please use GEN-UNIQUE or GEN-UNIQUE-[some-number]

gatneil commented on rds-deployment-existing-ad/azuredeploy.parameters.json:
```diff
@@ -2,32 +2,11 @@
   "contentVersion": "1.0.0.0",
   "parameters": {
+    "dnsLabelPrefix": {
+      "value": "rdsuniquedns3"
```
GEN-UNIQUE

gatneil commented on rds-deployment/azuredeploy.parameters.json:
```diff
@@ -3,19 +3,10 @@
-    },
-    "adminUsername": {
-      "value": "azureUser"
+      "value": "aduniqedns2"
```
GEN-UNIQUE

gatneil commented:
Please address the above comments

MahendraAgrawal commented:
Hi Neil,
I have fixed most of the feedback from below, checked in. I updated api versions to raw strings and use GEN-UNIQUE for all three templates and their respective parameter files. Here are some clarifications. Please note that our goal is to make templates consistent with Azure and AzureStack. These updated templates meet that goal. Please help merge this change.

·        The default password in existing-ad template is for automation purpose. We run them on daily basis as a KPI measurement for AzureStack.

·        We want to keep assetlocation as is because if I parameterize it, this will break dependent templates of partners and also require lot more changes on our automation.

·        The admin cred does goto protected settings. Just look at reference in next block.
Thanks a lot for help,
Mahendra.

From: Neil Sant Gat [mailto:notifications@github.com]
Sent: Friday, September 30, 2016 2:51 PM
To: Azure/azure-quickstart-templates <azure-quickstart-templates@noreply.github.com>
Cc: Mahendra Agrawal <mahendra@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-quickstart-templates] Updating parameters. (#2564)


@gatneil requested changes on this pull request.

________________________________

In rds-deployment-custom-image-rdsh/azuredeploy.json<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-quickstart-templates%2Fpull%2F2564%23pullrequestreview-2406872&data=01%7C01%7Cmahendra%40microsoft.com%7C5b84127dfeb24111744a08d3e97bcf55%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=4jgrqDVvcKNb56gTHLeoXXFOLsXeeMhMLfRuluvtigs%3D&reserved=0>:

>      },

-  "resources": [

-    {

-      "apiVersion": "[variables('apiVersion')]",

-      "type": "Microsoft.Network/publicIPAddresses",

-      "name": "[variables('publicIpRef')]",

-      "location": "[resourceGroup().location]",

-      "properties": {

-        "publicIPAllocationMethod": "Dynamic",

-        "dnsSettings": {

-          "domainNameLabel": "[parameters('dnsLabelPrefix')]"

+    "resources": [

+      {

+        "apiVersion": "[variables('apiVersionNRP')]",

please use raw strings for apiVersion instead of variables

________________________________

In rds-deployment-custom-image-rdsh/azuredeploy.json<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-quickstart-templates%2Fpull%2F2564%23pullrequestreview-2406872&data=01%7C01%7Cmahendra%40microsoft.com%7C5b84127dfeb24111744a08d3e97bcf55%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=4jgrqDVvcKNb56gTHLeoXXFOLsXeeMhMLfRuluvtigs%3D&reserved=0>:

>          "subnet-id": "[concat(resourceId('Microsoft.Network/virtualNetworks',parameters('adVnetName')),'/subnets/',parameters('adSubnetName'))]",

         "publicIpRef": "publicIp",

+        "brokerIpRef": "brokerpublicIp",

+        "gatewayIpRef": "gatewaypublicIp",

         "assetLocation": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/rds-deployment/",

Please see the best practice around _artifactsLocation and _artifactsLocationSasToken: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-quickstart-templates%2Fblob%2Fmaster%2F1-CONTRIBUTION-GUIDE%2Fbest-practices.md&data=01%7C01%7Cmahendra%40microsoft.com%7C5b84127dfeb24111744a08d3e97bcf55%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=wL0iv6btwQiKYcwKsSZznJbQNpTP2HzBwkjhWwuycrc%3D&reserved=0>

________________________________

In rds-deployment-custom-image-rdsh/azuredeploy.json<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-quickstart-templates%2Fpull%2F2564%23pullrequestreview-2406872&data=01%7C01%7Cmahendra%40microsoft.com%7C5b84127dfeb24111744a08d3e97bcf55%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=4jgrqDVvcKNb56gTHLeoXXFOLsXeeMhMLfRuluvtigs%3D&reserved=0>:

> +      },

+      "defaultValue": "[concat('ADStaticSubnet',resourceGroup().name)]"

+    },

+    "adminUsername": {

+      "type": "string",

+      "metadata": {

+        "description": "The name of the administrator of the new VM and the domain. Exclusion list: 'administrator'. For example johnadmin"

+      },

+      "defaultValue": "vmadmin"

+    },

+    "adminPassword": {

+      "type": "securestring",

+      "metadata": {

+        "description": "The password for the administrator account of the new VM and the domain"

+      },

+      "defaultValue": "[concat('Subscription#',subscription().subscriptionId)]"

are we sure we want a default password?

________________________________

In rds-deployment-custom-image-rdsh/azuredeploy.json<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-quickstart-templates%2Fpull%2F2564%23pullrequestreview-2406872&data=01%7C01%7Cmahendra%40microsoft.com%7C5b84127dfeb24111744a08d3e97bcf55%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=4jgrqDVvcKNb56gTHLeoXXFOLsXeeMhMLfRuluvtigs%3D&reserved=0>:

> +                    "name": "[concat('rdsh-', copyindex(),'/sessionhost')]",

+                    "location": "[resourceGroup().location]",

+                    "dependsOn": [

+                        "[resourceId('Microsoft.Compute/virtualMachines', concat('rdsh-', copyindex()))]"

+                    ],

+                    "properties": {

+                        "publisher": "Microsoft.Powershell",

+                        "type": "DSC",

+                        "typeHandlerVersion": "2.11",

+                        "autoUpgradeMinorVersion": true,

+                        "settings": {

+                            "ModulesUrl": "[concat(variables('assetLocation'),'/Configuration.zip')]",

+                            "ConfigurationFunction": "Configuration.ps1\\SessionHost",

+                            "Properties": {

+                                "DomainName": "[parameters('adDomainName')]",

+                                "AdminCreds": {

Shouldn't AdminCreds go in protectedSettings?

________________________________

In rds-deployment-custom-image-rdsh/azuredeploy.parameters.json<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-quickstart-templates%2Fpull%2F2564%23pullrequestreview-2406872&data=01%7C01%7Cmahendra%40microsoft.com%7C5b84127dfeb24111744a08d3e97bcf55%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=4jgrqDVvcKNb56gTHLeoXXFOLsXeeMhMLfRuluvtigs%3D&reserved=0>:

>      },

     "dnsLabelPrefix": {

-      "value": "GEN-UNIQUE"

+      "value": "rduniquedns1"

please use GEN-UNIQUE or GEN-UNIQUE-[some-number]

________________________________

In rds-deployment-custom-image-rdsh/azuredeploy.parameters.json<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-quickstart-templates%2Fpull%2F2564%23pullrequestreview-2406872&data=01%7C01%7Cmahendra%40microsoft.com%7C5b84127dfeb24111744a08d3e97bcf55%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=4jgrqDVvcKNb56gTHLeoXXFOLsXeeMhMLfRuluvtigs%3D&reserved=0>:

>      "rdshImageStorageAccountName": {

-      "value": "Provide-StorageName"

+      "value": "imageuniquestorename1"

please use GEN-UNIQUE or GEN-UNIQUE-[some-number]

________________________________

In rds-deployment-existing-ad/azuredeploy.parameters.json<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-quickstart-templates%2Fpull%2F2564%23pullrequestreview-2406872&data=01%7C01%7Cmahendra%40microsoft.com%7C5b84127dfeb24111744a08d3e97bcf55%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=4jgrqDVvcKNb56gTHLeoXXFOLsXeeMhMLfRuluvtigs%3D&reserved=0>:

> @@ -2,32 +2,11 @@

   "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",

   "contentVersion": "1.0.0.0",

   "parameters": {

+    "dnsLabelPrefix": {

+      "value": "rdsuniquedns3"

GEN-UNIQUE

________________________________

In rds-deployment/azuredeploy.parameters.json<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-quickstart-templates%2Fpull%2F2564%23pullrequestreview-2406872&data=01%7C01%7Cmahendra%40microsoft.com%7C5b84127dfeb24111744a08d3e97bcf55%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=4jgrqDVvcKNb56gTHLeoXXFOLsXeeMhMLfRuluvtigs%3D&reserved=0>:

> @@ -3,19 +3,10 @@

   "contentVersion": "1.0.0.0",

   "parameters": {

     "dnsLabelPrefix": {

-      "value": "GEN-UNIQUE"

-    },

-    "adDomainName": {

-      "value": "mydomain.local"

-    },

-    "adminUsername": {

-      "value": "azureUser"

+      "value": "aduniqedns2"

GEN-UNIQUE

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-quickstart-templates%2Fpull%2F2564%23pullrequestreview-2406872&data=01%7C01%7Cmahendra%40microsoft.com%7C5b84127dfeb24111744a08d3e97bcf55%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=4jgrqDVvcKNb56gTHLeoXXFOLsXeeMhMLfRuluvtigs%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAOdkawmKA2JFn6vJf96Hb9SJ62PJZN9jks5qvYQqgaJpZM4KKZT2&data=01%7C01%7Cmahendra%40microsoft.com%7C5b84127dfeb24111744a08d3e97bcf55%7C72f988bf86f141af91ab2d7cd011db47%7C1&sdata=07N%2BWn23bv7aDhtiwZ96FLe8PHdRyBcSAHB5%2BESK4wU%3D&reserved=0>.

gatneil commented on rds-deployment-custom-image-rdsh/azuredeploy.json:
I'm not sure i recommend using apiVersions[0] (as opposed to choosing an apiVersion).

gatneil commented:
Formatting is a bit odd in some places (left a comment saying "formatting" in one such place), and I'm not sure I recommend using apiVersions[0] (as opposed to choosing an apiVersion). Please address these in a subsequent PR.
